tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/subbuilder/SetBuilder.scala

```diff
@@ -44,21 +43,35 @@ trait SetBuilder extends UnsafeSetBuilder {
 
   /** { x \in set: p } */
   def filter(x: TBuilderInstruction, set: TBuilderInstruction, p: TBuilderInstruction): TBuilderInstruction =
-    for {
-      xEx <- x
-      setEx <- set
-      pEx <- p
-      _ <- markAsBound(xEx)
-    } yield _filter(xEx, setEx, pEx)
+    boundVarIntroductionTernary(_filter)(x, set, p)
 
   /** { mapExpr: x1 \in set1 , ..., xN \in setN }, must have at least 1 var-set pair */
   def map(
       e: TBuilderInstruction,
       varSetPairs: (TBuilderInstruction, TBuilderInstruction)*): TBuilderInstruction = {
-    val args = varSetPairs.flatMap { case (k, v) =>
-      Seq(k, v)
-    }
-    mapMixed(e, args: _*)
+    for {
+      mapExpr <- e
+      boundAfterMapExpr <- allBound // xi may not appear as bound in mapExpr
```

Comment:
Again, I don't understand what this comment is doing here. Doesn't seem related to this line and there is no xi mentioned anywhere.

Reply:
`xi` refers to
https://github.com/informalsystems/apalache/blob/3c9285aedfb2acb11294008f67e5406d20faca3a/tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/subbuilder/SetBuilder.scala#L48
This is actually why the code in `BuilderUtil` mentions `x`, since I originally wrote it for `forall` (`\A x \in S: P`) and then moved it to the utility class when I noticed it could be reused.

Reply:
Ah that makes sense. So the comment was a bit not of date. Thanks for the explanation and cleanup.